features/dd-sdk-android-trace-api/src/main/kotlin/com/datadog/android/trace/api/DatadogTracingConstants.kt

```diff
@@ -0,0 +1,217 @@
+ * Copyright 2016-Present Datadog, Inc.
+ */
+package com.datadog.android.trace.api
+
+/**
+ * Contains constants related to Datadog tracing. This object groups constants into several
+ * nested objects for better organization, such as [Tags], [PrioritySampling],
+ * [TracerConfig], [LogAttributes], and [ErrorPriorities].
+ */
+object DatadogTracingConstants {
+    /**
+     * Represents the default value for enabling or disabling asynchronous context propagation
+     * in Datadog's tracing implementation.
+     */
+    const val DEFAULT_ASYNC_PROPAGATING: Boolean = true
+
+    /**
+     * Contains constants used for tagging spans in the Datadog tracer.
+     */
+    object Tags {
+        /**  The URL of the HTTP request. */
+        const val KEY_HTTP_URL: String = "http.url"
+
+        /**
+         * Represents the key used for tagging the kind of a span in the Datadog tracer.
+         * It is used to indicate the role or type of a span, such as client, server, producer, or consumer.
+         */
+        const val KEY_SPAN_KIND: String = "span.kind"
+
+        /** Indicates the desired action to be performed for a given resource. */
+        const val KEY_HTTP_METHOD: String = "http.method"
+
+        /** The HTTP response status code. */
+        const val KEY_HTTP_STATUS: String = "http.status_code"
+
+        /** String representing the error message. */
+        const val KEY_ERROR_MSG: String = "error.msg"
+
+        /** String representing the type of the error. */
+        const val KEY_ERROR_TYPE: String = "error.type"
+
+        /** human readable version of the stack. */
+        const val KEY_ERROR_STACK: String = "error.stack"
+
+        /**
+         * Represents the value used for tagging a span as a "client" in the Datadog tracer with [KEY_SPAN_KIND] key.
+         */
+        const val VALUE_SPAN_KIND_CLIENT: String = "client"
+
+        /**
+         * Represents the value used for tagging a span as a "server" in the Datadog tracer with [KEY_SPAN_KIND] key.
+         */
+        const val VALUE_SPAN_KIND_SERVER: String = "server"
+
+        /**
+         * Represents the value used for tagging a span as a "producer" in the Datadog tracer.
+         */
+        const val VALUE_SPAN_KIND_PRODUCER: String = "producer"
+
+        /**
+         * Represents the value used for tagging a span as a "consumer" in the Datadog tracer.
+         */
+        const val VALUE_SPAN_KIND_CONSUMER: String = "consumer"
+
+        /**
+         * Represents the key used for tagging the analytics sample rate in the Datadog tracer.
+         */
+        const val KEY_ANALYTICS_SAMPLE_RATE: String = "_dd1.sr.eausr"
+
+        /**
+         * Represents the resource name tag used for tracing spans.
+         */
+        const val RESOURCE_NAME: String = "resource.name"
+
+        /**
+         *  String representing the error message.
+         */
+        const val ERROR_MSG: String = "error.message"
```

Comment:
I see that we have both `ERROR_MSG` and `KEY_ERROR_MSG`, and also `ERROR_TYPE` and `KEY_ERROR_TYPE` with the same values. Is this on purpose?

Reply:
ERROR_MSG - it's part of dd-trace-java. 
KEY_ERROR_MSG -part of public API for internal trace layer. 

Yes, it'n on purpose, I want to keep public only those constants that is being used by other modules in the SDK,

Reply:
thanks for this catch, yep, was renamed during one of review fixes. Aligned now